lib/arbitrary.js

```diff
@@ -150,6 +150,61 @@ function oneof() {
   });
 }
 
+// Return a lazy arbitrary that delegates to another arbitrary at its
+// 'strict' property. An arbitrary must be assigned to that property before
+// this arbitrary can generate anything.
+function lazyArbitrary() {
+  var arb = {};
+  // This function must be pure because it will not be called with
+  // meaningful context.
+  arb.generator = generator.bless(function (size) {
+    return arb.strict.generator(size);
+  });
+  arb.shrink = shrink.noop;
+  arb.show = show.def;
+  arb = arbitraryBless(arb);
+  return arb;
+}
+
+/**
+  - `letrec(
+       (tie: key -> (arbitrary a | arbitrary b | ...))
+       -> { key: arbitrary a, key: arbitrary b, ... }):
+     { key: arbitrary a, key: arbitrary b, ... }`
+
+     Mutually recursive definitions. Every reference to a sibling arbitrary
+     should go through the `tie` function.
+
+     `{ arb1, arb2 } = jsc.letrec(function (tie) {
+       return {
+         arb1: jsc.tuple(jsc.int, jsc.oneof(jsc.const(null), tie("arb2"))),
+         arb2: jsc.tuple(jsc.bool, jsc.oneof(jsc.const(null), tie("arb1"))),
+       }
+     });`
+*/
+function letrec(definition) {
```

Comment:
smart. I see. The lazyArbs trick is because we don't know the key set before calling `definition`.